src/Hooks/usePokemon.js

```diff
@@ -0,0 +1,38 @@
+import { fetchPokemon, fetchPokeTypes } from '../services/fetch-utils';
+import { useState, useEffect } from 'react';
+
+export default function usePokemon() {
+  const [pokemonList, setPokemonList] = useState([]);
+  const [type, setType] = useState('All');
+  const [typeList, setTypeList] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [inputValue, setInputValue] = useState('');
+  useEffect(() => {
+    const loadData = async () => {
+      try {
```

Comment:
With the way you have this implemented you should be sure to set your loading state to `true` just before calling the fetch so that subsequent requests (when type changes) will also display the loading <div>

With the way it is now loading gets set to `false` the first time the page loads, but will never again be set to `true`